crates/aiken-lang/src/format.rs

```diff
@@ -549,6 +557,63 @@ impl<'comments> Formatter<'comments> {
             .append("}")
     }
 
+    fn definition_validator<'a>(
+        &mut self,
+        params: &'a [UntypedArg],
+        fun: &'a UntypedFunction,
+        end_position: usize,
+    ) -> Document<'a> {
+        // Fn and args
+        let head = "fn".to_doc().append(wrap_args(
+            fun.arguments.iter().map(|e| (self.fn_arg(e), false)),
+        ));
+
+        // Add return annotation
+        let head = match &fun.return_annotation {
+            Some(anno) => head.append(" -> ").append(self.annotation(anno)),
+            None => head,
+        }
+        .group();
+
+        // Format body
+        let body = self.expr(&fun.body);
+
+        // Add any trailing comments
+        let body = match printed_comments(self.pop_comments(fun.end_position), false) {
+            Some(comments) => body.append(line()).append(comments),
+            None => body,
+        };
+
+        // validator name(params)
+        let v_head = "validator"
+            .to_doc()
+            .append(" ")
+            .append(fun.name.as_str())
+            .append(if !params.is_empty() {
+                wrap_args(params.iter().map(|e| (self.fn_arg(e), false)))
+            } else {
+                "".to_doc()
+            });
+
+        // Stick it all together
+        let inner_fn = head
+            .append(" {")
+            .append(line().append(body).nest(INDENT).group())
+            .append(line())
+            .append("}");
+
+        let inner_fn = match printed_comments(self.pop_comments(end_position), false) {
+            Some(comments) => inner_fn.append(line()).append(comments),
+            None => inner_fn,
+        };
+
+        v_head
+            .append(" {")
+            .append(line().append(inner_fn).nest(INDENT).group())
+            .append(line())
+            .append("}")
+    }
```

Comment:
Could deserve a some quick tests in `tests/format.rs`

Reply:
cool yea, I'll probably also add some check test too